sources/rest_api/detector.py

```diff
@@ -0,0 +1,82 @@
+from dlt.sources.helpers.requests import Response
+
+from .paginators import HeaderLinkPaginator, JSONResponsePaginator
+
+RECORD_KEY_PATTERNS = {'data', 'items', 'results', 'entries'}
+NEXT_PAGE_KEY_PATTERNS = {'next', 'nextpage', 'nexturl'}
+
+
+def find_records_key(dictionary, path=None):
```

Comment:
Maybe we can restrict `dictionary` using typing?
<img width="610" alt="image" src="https://github.com/dlt-hub/verified-sources/assets/354868/09153247-6fff-4336-b9d4-2ecc242db73f">
https://replit.com/@sultaniman/restricted-dict-keys-typing?v=1